distributed/diagnostics/bokeh/status/main.py

```diff
@@ -35,4 +39,14 @@ def task_update():
         task_table_update(task_source, msg)
 doc.add_periodic_callback(task_update, messages['tasks']['interval'])
 
-doc.add_root(vplot(worker_table, task_table))
+resource_source, resource_plot = resource_profile_plot()
+@gen.coroutine
+def resource_update():
+    with log_errors():
+        yield messages['workers']['condition'].wait()
+        worker_buffer = messages['workers']['deque']
+        times_buffer = messages['workers']['times']
```

Comment:
Might want to call `list` on both of these two deques before passing them down to another function, just for cleanliness's sake.